### ISSUE TITLE
model.num_timesteps or a similar method inside a SubprocVecEnv

### ISSUE THREAD

**Issue by olyanos** [hill-a/stable-baselines#1163]
Hi,
I am not sure if this is the right place to ask this question but I just can not seem to find the answer elsewhere.
I currently have a parallel custom environment that I am calling using 
I have already registered the environment and it's working as expected the PPO algorithm produces good results

```python
env = SubprocVecEnv([make_env(i) for i in range(number_of_envs)])
env = VecCheckNan(env, raise_exception=True)
env = VecMonitor(env)
```

**Question**
How can I get the number of total time-steps the algorithm has reached inside the step() function in my environment?
i.e I want to know which RL time-step (globaly across all envs) I am currently in at this time-step.
I tried to set a counter inside the step() function but this counter would be only local to that environment not a global one that computes how many steps the algorithm took.
I have been trying to log at which time-steps the PPO algorithm actually end up finding a solution to my problem (Good termination)

I know that outside the gym environment I can call 
```python
model.num_timesteps
```
after I initialized my model as:
```python
model = PPO('MlpPolicy', env,....)
```
However, I am unable to use model.num_timesteps inside the gym environment since it is not defined in that module

**System Info**
Tensorflow version = 2.8.0
Keras version = 2.8.0
Stable baseline version = 1.2.0
OpenAI Gym version = 0.19.0
Torch version = 1.11.0
Python version = 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]
NVIDIA GeForce RTX 3090.

**Comment by Miffyli** [comment]
There is no support for this and it sounds like a weird use (does not sound like it would fit in normal RL frameworks, so I suspect you will have poor results with your experiments). I recommend you manually track steps inside your environments (as you said), but multiply it by the number of environments you create. I.e., pass `n_envs` to your environments as well, and track your steps as `n_steps += n_envs` in each env.

**Comment by olyanos** [comment]
Thank you @Miffyli for your reply. I know it is not a typical use of the times-steps counter but I needed it in order to log the first time-step the PPO algorithm actually succeed in my environment and monitor the progress of the agent's learning with time (in addition to the reward vs time-step plot).

My code does work well with my custom environment and SB3 PPO produces good results. I just wanted to plot time-steps vs successful termination. 

In terms of your suggested solution, I do not think it would be accurate as my parrallel envs are not necessarily synchronous at all times. I.e depending on the action taken by the agent some environments will take longer to compute (i.e longer time in step() ) than others, thus by adding +n_envs at each step() may not produce accurate correct numbers.
Sorry I know my environment is complex :( but so is the problem I am working on.

I was able to log the first success the agent finds by throwing an Exception:
if success:
```python
raise ValueError('Good termination')
```
Then from my main file
I called:
```python
     try:
         model.learn(total_timesteps=RL_total_time_steps)
     except Exception as e:
         timesteps_ctr = model.num_timesteps
         print('Total Timesteps until first success= '+str(timesteps_ctr))
```
But this would only log the first success not the rest of them.

Thank you for your help, I will close this issue now and update if I find a solution.
